File: week06-Python/pset6/cash/cash.py
from cs50 import get_float


def main():
    # define variables
    quarters = 25
    dimes = 10
    nickels = 5
    pennies = 1
    cents = 0  # to be used in converting the entry of the user to cents
    coins = 0  # Counter for coins

    dollars = get_positive_get_float()

    cents = int(dollars * 100)

    # loop through the cents and calculate the remaining 
    while cents >= quarters:
        cents -= quarters
        coins += 1
    while cents >= dimes:
        cents -= dimes
        coins += 1
    while cents >= nickels:
        cents -= nickels
        coins += 1
    while cents >= pennies:
        cents -= pennies
        coins += 1

    print(coins)


# function to get a float from the user
def get_positive_get_float():
    while True:
        n = get_float("Change owed: ")
        if n > 0:
            break
    return n


main()

File: week01-C/pset1/cash.c
/*pset1 , cash less comfortable ,
Program will print the lease amount of coins given as a change
created by Ammar Sayed , 4th Aug 2019
*/
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int quarters = 25;
int dimes = 10;
int nickels = 5;
int pennies = 1;
float dollars;
int cents ;
int coins;

int main(void)
{

    // prompting the user for the change in dollars
    do
    {
        dollars = get_float("change: \n");
    }
    while (dollars < 0);

    //Converting the dollars to cents and round the number
    cents = round(dollars * 100);

    //Testing the input
    printf("change is %i cents\n", cents);

    //checking the quarters
    while (cents >= quarters)
    {
        coins++;
        cents = cents - quarters;
    }

    //checking the dimes
    while (cents >= dimes)
    {
        coins++;
        cents = cents - dimes;
    }

    //checking the nickels
    while (cents >= nickels)
    {
        coins++;
        cents = cents - nickels;
    }

    //checking the pennies
    while (cents >= pennies)
    {
        coins++;
        cents = cents - pennies;
    }

    //Print the number of coins calcualted
    printf("coins: %i\n", coins);

}